src/components/Commentable/index.tsx

```diff
@@ -1,29 +1,45 @@
 import * as React from 'react';
 import makeClassName from 'classnames';
 
+import AddComment, { PublicProps as AddCommentProps } from './AddComment';
 import { AnyReactNode } from '../../typeUtils';
 import styles from './styles.module.scss';
 
 export type PublicProps = {
+  addCommentClassName?: string;
   as: React.ReactType;
-  children?: AnyReactNode;
+  children: (addComment: JSX.Element) => AnyReactNode;
```

Comment:
As far as I can tell it's not possible, at least not with current [TypeScript support for JSX](https://www.typescriptlang.org/docs/handbook/jsx.html#the-jsx-result-type). I guess that makes sense since you can't manipulate JSX from React components anyway (i.e. it doesn't matter what component it's an instance of).